core/datatrak.c

```diff
@@ -561,11 +578,105 @@ static unsigned char *dt_dive_parser(unsigned char *runner, struct dive *dt_dive
 	free(devdata);
 	return NULL;
 }
+/*
+ * Parses the header of the .add file, returns the number of dives in
+ * the archive (must be the same than number of dives in .log file).
+ */
+static unsigned int wlog_header_parser (struct memblock *mem)
+{
+	int tmp;
+	unsigned char *runner = (unsigned char *) mem->buffer;
+	if (!runner)
+		return -1;
```

Comment:
Return `-1` in an `unsigned int` function. That's obscure..?

Reply:
The function doesn't really need to be unsigned as I don't expect anyone to have such a big number of dives logged. Returning a negative value on error is mandatory, BTW, as 0 and positives can be expected correct values.